src/Panel.vue

```diff
@@ -329,6 +332,14 @@
         if (this.hasSrc && (this.preloadBool || this.expandedBool)) {
           this.$refs.retriever.fetch()
         }
+
+        if (this.hasCustomHeader) {
+          const headerInnerHTML = this.$refs.headerWrapper.innerHTML;
+          const wrappedElementName = jQuery(headerInnerHTML).attr("name");
+          this.$refs.headerWrapper.innerHTML =
+            jQuery(headerInnerHTML).unwrap().prepend(this.caretHtml + ' ')
+              .wrap(`<${wrappedElementName}></${wrappedElementName}>`).parent().html();
```

Comment:
Should abstract this to a method for code quality and maintainability. Maybe `insertCaretInsideCustomHeader`?